src/runtime.rs

```diff
@@ -63,6 +64,28 @@ pub fn cnb_runtime<P: Platform, BM: DeserializeOwned, E: Debug + Display>(
     if let Err(lib_cnb_error) = result {
         exit(error_handler.handle_error(lib_cnb_error));
     }
+
+    match read_buildpack_toml::<BM, E>() {
```

Comment:
If I'm reading this correctly, this new check runs after the buildpack's detect/build runs, rather than before. What are your thoughts on moving this before? (To fail early and also prevent confusing error messages from usages of libcnb within detect/build due to something buildpack API related [though struggling to think of examples of when that might occur?])

Reply:
You're right, and this wasn't the intention. I'll fix this. :)

Reply:
Landed in 67d4bc6